Hooks: Add hook for wx._core

BrendanSimon opened the pull request:
wxPython 4.0.7 doesn't seem to find wx._core.
This hook addresses this issue.

htgoebel commented:
Thank you for this pull-request. 

Can you please explain a bit, why collect_data_files is used instead of collect_submodules?

Please remove the unused code in the hook.

For this pull-request to be merged, please also submit a [**changelog entry**](https://pyinstaller.readthedocs.io/en/latest/development/changelog-entries.html) so our users can learn about your change.

Thanks.

BrendanSimon commented:
The reason `collect_data_files` is used instead of `collect_submodules` is because I just copied those from another hook that I thought was similar, and it "solved my problem".

I will take another look at it and try `collect_submodules` and resubmit the PR.

Thanks for the feedback :)

BrendanSimon commented:
I tried `hiddenimports = collect_submodules('wx._core')` and got the following error at run-time.

```
    exec(bytecode, module.__dict__)
  File "lib\site-packages\wx\core.py", line 12, in <module>
ImportError: DLL load failed: The parameter is incorrect.
```

I also tried `binaries = collect_dynamic_libs('wx._core')` but it failed too with the same error message.

I went back to scratch and removed the `--strip` and added `--noupx` from by build script.  Now it works (runs) without any hook required !!

Adding back `--strip` and keeping `--noupx` => works/runs ok.

I had a UPX exlclusion for `vcruntime140.dll`, but with trial and error I discovered that adding `msvcp140.dll` to the upx exclusion fixed the `wx._core` issue (at least with a very simple test app).

Looks like the above PR is bogus.

BrendanSimon commented:
Also confirmed with real non-trivial wxPython application.

So I'm going to close this PR.

BrendanSimon commented:
Unless there is some way to add items to the upx exclude list via the hooks mechanism ???
